### ISSUE TITLE
Support for `/external:I` experimental option from MSVC

### ISSUE THREAD

**Issue by poppolopoppo** [fastbuild/fastbuild#764]
Hello!

We would like to use the experimental feature `/experimental:external` from [Visual Studio 2017 version 15.6 Preview 1](https://devblogs.microsoft.com/cppblog/broken-warnings-theory/).
This option enables `/external:I`, an alternative to `/I` to specify an include directory, which is useful for handling compilation warnings in external code when used together with `/external:W[n]`.

From what I know FASTBuild is already aware of `/I` for `cl.exe`, which allows it to detect, transform or strip those arguments according to build phase; and stripping the include paths seems to be a required mechanism to enable distributed compilation of preprocessed sources for this compiler. 

But apparently this mechanism isn't available for the option `/external:I`, as FASTBuild doesn't know of this feature.
When I ran a BFF config using this flag I got the following issues for distributed jobs:
````shell
c1xx: warning C5044: An argument to command-line option /external:I points to a path 'c:\program files (x86)\windows kits\netfxsdk\4.6.2\include\um' that does not exist
c1xx: warning C5044: An argument to command-line option /external:I points to a path 'c:\program files (x86)\windows kits\10\include\10.0.18362.0\ucrt' that does not exist
c1xx: warning C5044: An argument to command-line option /external:I points to a path 'c:\program files (x86)\windows kits\10\include\10.0.18362.0\shared' that does not exist
c1xx: warning C5044: An argument to command-line option /external:I points to a path 'c:\program files (x86)\windows kits\10\include\10.0.18362.0\um' that does not exist
c1xx: warning C5044: An argument to command-line option /external:I points to a path 'c:\program files (x86)\windows kits\10\include\10.0.18362.0\winrt' that does not exist
````
I can confirm that disabling the distribution fixed those.

I think similar issues might happen with `/imsvc`/`-imsvc` from `clang-cl.exe` [LLVM driver](https://llvm.org/builds/), which we also use by forcing the compiler family to MSVC (which is sad since explicit `clang-cl` support could be a [win](https://fastbuild.org/docs/features/caching.html), but this is another topic ;) ).

Do you think FASTBuild should support those new flags ?

Thanks!

**Comment by ffulin** [comment]
FASTBuild should support these flags. Updating the main compiler invocation to remove the new option when needed should be trivial.

A little more work will be required to correctly process these paths in the LightCache, but that should still be quite manageable.

**Comment by ffulin** [comment]
I spent some time investigating this issue, and while adding support to detect those args (as per your PR) was fairly straight forward, I found that it doesn't actually help, because:
a) On a remote machine, if the directory doesn't exist, compilation fails (assuming /WX is used)
b) On a remote machine, if the directory does exist, /external:I doesn't work (seems to not respect #line directives)

It doesn't seem there is a way for this feature to be used with FASTBuild in its current form.

I've raised a Developer Community issue to see if these issues can be fixed in the compiler: https://developercommunity2.visualstudio.com/t/experimental:external-does-not-work-wit/1313253

**Comment by ffulin** [comment]
The issue i originally raised in the Developer Community was incorrectly closed as a duplicate, but I re-reported the issue, split in two:
* https://developercommunity2.visualstudio.com/t/external:I-requires-directory-to-exist/1340525
* https://developercommunity2.visualstudio.com/t/experimental:I-incorrectly-ignores-dire/1306689

It seems those issues have now been closed as 'Fixed In: Visual Studio 2019 version 16.10'. Pending verification of those fixes, we can hopefully resolve this. (Some or perhaps even all of the actual necessary behavior on the FASTBuild side has already been implemented)

**Comment by ffulin** [comment]
I think this issue slipped through the cracks a little, but FASTBuild (as of v1.08 at least, but probably earlier) is working correctly with /external:I command lines as long as the compiler has the appropriate fixes mentioned above. (Latest 2019 or 2022 should work)
